src/Security/Voter/ParticipationVoter.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Security\Voter;
+
+use App\Entity\Participation;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class ParticipationVoter extends Voter
+{
+    protected function supports(string $attribute, $subject): bool
+    {
+        return \in_array($attribute, ['SHOW_PARTICIPATION'])
+            && ($subject instanceof Participation);
+    }
+
+    /**
+     * @param Participation $subject
+     */
+    protected function voteOnAttribute(string $attribute, $subject, TokenInterface $token): bool
+    {
+        $user = $token->getUser();
+        if (!$user instanceof UserInterface) {
+            return false;
+        }
+
+        if (\in_array('ROLE_ADMIN', $user->getRoles(), true)) {
+            return true;
+        }
```

Comment:
You'll need to use the RoleHierarchy on this one. See https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/Security/Core/Authorization/Voter/RoleHierarchyVoter.php